partner_parent_tag/models/res_partner.py

```diff
@@ -0,0 +1,22 @@
+# Copyright 2025 Ángel Rivas <angel.rivas@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model_create_multi
+    def create(self, vals_list):
```

Comment:
Contacts should be created at the beginning and then edit those whose parent contact has categories. They way you've done it would not work if you create the parent and child contacts at the same time.